webpack/assets/javascripts/hosts/tableCheckboxes.js

```diff
@@ -0,0 +1,254 @@
+    n__('Select this host', 'Select all<b> %s </b> hosts', pagination.total),
+    pagination.total
+  );
+  const multpleAlert = $('#multiple-alert');
+  multpleAlert
+    .find('.text')
+    .html(
+      `${alertText} <a href="#" onclick="tfm.hosts.table.multipleSelection();">${selectText}</a>`
+    );
+  multpleAlert.data('multiple', false);
+  $('.select_count').html(pagination.perPage);
+}
+
+export function toggleCheck() {
+  const pagination = paginationMetaData();
+  const multpleAlert = $('#multiple-alert');
+  const checked = $('#check_all').is(':checked');
+  $('.host_select_boxes').each((index, box) => {
+    box.checked = checked;
+    hostChecked(box);
+  });
+  if (checked && pagination.perPage - pagination.total < 0) {
+    multpleAlert.show('slow');
+    multpleAlert.data('multiple', false);
+  } else if (!checked) {
+    multpleAlert.hide('slow');
+    multpleAlert.data('multiple', false);
+    cleanHostsSelection();
+  }
+  return false;
+}
+
+export function toggleMultipleOkButton({ value }) {
+  const btn = $('#confirmation-modal .btn-primary');
+  if (value !== 'disabled') btn.removeClass('disabled').attr('disabled', false);
+  else btn.addClass('disabled').attr('disabled', true);
+}
+
+export function submitModalForm() {
+  if (!$('#keep_selected').is(':checked')) removeForemanHostsCookie();
+  if (isMultple()) {
+    const query = $('<input>')
+      .attr('type', 'hidden')
+      .attr('name', 'search')
+      .val($('#search').val());
+    $('#confirmation-modal form').append(query);
+  }
+  $('#confirmation-modal form').submit();
+  $('#confirmation-modal').modal('hide');
+}
+
+function isMultple() {
+  return $('#multiple-alert').data('multiple');
+}
+
+function getBulkParam() {
+  return isMultple()
+    ? { search: $('#search').val() }
+    : { host_ids: foremanSelectedHosts };
+}
+
+export function buildModal(element, url) {
+  const data = getBulkParam();
+  const title = $(element).attr('data-dialog-title');
+  $('#confirmation-modal .modal-header h4').text(title);
+  $('#confirmation-modal .modal-body')
+    .empty()
+    .append("<div class='modal-spinner spinner spinner-lg'></div>");
+  $('#confirmation-modal').modal();
+  $('#confirmation-modal .modal-body').load(
+    `${url} #content`,
+    data,
+    (response, status, xhr) => {
+      $('#loading').hide();
+      $('.modal-body #content').show();
+      $('#submit_multiple').val('');
+      if (isMultple()) $('#multiple-modal-alert').show();
+      const b = $('#confirmation-modal .btn-primary');
+      if ($(response).find('#content form select').length > 0)
+        b.addClass('disabled').attr('disabled', true);
+      else b.removeClass('disabled').attr('disabled', false);
+    }
+  );
+  return false;
+}
+
+export function buildRedirect(url) {
+  const data = getBulkParam();
+  const uri = `${url}?${$.param(data)}`;
+  window.location.replace(uri);
+}
+
+function paginationMetaData() {
+  const pagination = $('#pagination');
```

Comment:
@glekner @amirfefer this seems to be broken after pagination refactoring, its outside the scope of this PR, so its mostly FYI as I've encountered it.

Reply:
Do we have a redmine issue to fix this?

Reply:
https://projects.theforeman.org/issues/26494